src/main/java/apaintus/controllers/AttributeController.java

```diff
@@ -13,127 +15,88 @@
+        this.canvasController = this.controller.getCanvasController();
+
+        setSpinnerChangeListener();
+    }
+
+    @Override
+    public void initialize() {
+        createValueFactory();
+        fillSpinnerMap();
+    }
+
+    private void createValueFactory() {
+        shapeX.setValueFactory(new DoubleSpinnerValueFactory(0.0, Double.MAX_VALUE, 0.0, 1));
+        shapeY.setValueFactory(new DoubleSpinnerValueFactory(0.0, Double.MAX_VALUE, 0.0, 1));
+        shapeWidth.setValueFactory(new DoubleSpinnerValueFactory(0.0, Double.MAX_VALUE, 0.0, 1));
+        shapeHeight.setValueFactory(new DoubleSpinnerValueFactory(0.0, Double.MAX_VALUE, 0.0, 1));
+        shapeOrientation.setValueFactory(new DoubleSpinnerValueFactory(0.0, 360.0, 0.0, 1));
+    }
+
+    private void setSpinnerChangeListener() {
+        shapeX.valueProperty().addListener(canvasController.new SpinnerChangeListener(Attribute.COORDINATE_X));
+        shapeY.valueProperty().addListener(canvasController.new SpinnerChangeListener(Attribute.COORDINATE_Y));
+        shapeOrientation.valueProperty().addListener(canvasController.new SpinnerChangeListener(Attribute.ORIENTATION));
+        shapeWidth.valueProperty().addListener(canvasController.new SpinnerChangeListener(Attribute.WIDTH));
+        shapeHeight.valueProperty().addListener(canvasController.new SpinnerChangeListener(Attribute.HEIGHT));
+    }
+
+    private void fillSpinnerMap() {
+        spinners.put(Attribute.COORDINATE_X, shapeX.getValueFactory());
+        spinners.put(Attribute.COORDINATE_Y, shapeY.getValueFactory());
+        spinners.put(Attribute.ORIENTATION, shapeOrientation.getValueFactory());
+        spinners.put(Attribute.WIDTH, shapeWidth.getValueFactory());
+        spinners.put(Attribute.HEIGHT, shapeHeight.getValueFactory());
+    }
+
 //	public void bindTo(AnchorPane mainPane) {
 //		bar.prefWidthProperty().bind(mainPane.prefWidthProperty());
 //	}
-//
-//	public synchronized void update() {
-//		resetSpinners();
-//		Shape shape = SelectorService.getInstance().getSelectedShape();
-//		if (shape != null) {
-//			updateSpinners(shape);
-//		}
-//	}
-//
-//	public void resetSpinners() {
-//		for (Map.Entry<Attribute, SpinnerValueFactory<Double>> entry : spinners.entrySet()) {
-//			entry.getValue().setValue(0.0);
-//		}
-//	}
-//
-//	private void updateSpinners(Shape shape) {
-//		spinners.get(Attribute.X).setValue(shape.getCoordinates()[0]);
-//		spinners.get(Attribute.Y).setValue(shape.getCoordinates()[1]);
-//		spinners.get(Attribute.ORIENTATION).setValue(shape.getOrientation());
-//
-//		spinners.get(Attribute.WIDTH).setValue(shape.getBoundingBox().getWidth());
-//		spinners.get(Attribute.HEIGHT).setValue(shape.getBoundingBox().getHeight());
-//	}
 
-	public Controller getController() {
-		return controller;
-	}
+    public double getAttributeValue(Attribute attribute) {
+        return spinners.get(attribute).getValue();
+    }
+
+    public void update(DrawableShape shape) {
```

Comment:
is this normal that it is no longer synchonized?

Reply:
Il ne risque pas d'avoir de concurrence de thread ici puisque l'observer n'est plus là